### PR TITLE
Simplex updates

### PR DESCRIPTION
**Pull request by nfehn**
I am currently going through ExaDG to see which classes/functions are hard-coded for hypercube elements. In this PR,  asserts will be added for better usability. 

Extending the functionality would be part of future work, depending on the need and interest in certain functionality.

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]
Looks like a good step forward. I have some small notes, again not directly related to this PR but rather code beauty.

File: applications/incompressible_navier_stokes/fda_benchmark/include/grid.h [hunk 1]
```diff
@@ -387,7 +387,7 @@ create_grid_and_set_boundary_ids_nozzle(
         dealii::Point<dim> center = dealii::Point<dim>();
         manifold_vec[i] =
           std::shared_ptr<dealii::Manifold<dim>>(static_cast<dealii::Manifold<dim> *>(
-            new OneSidedCylindricalManifold<dim>(cell, face_ids[i], center)));
+            new OneSidedCylindricalManifold<dim>(*current_tria, cell, face_ids[i], center)));
```
[the remark this answers is not in the document]
Comment (nfehn) [reply applications/incompressible_navier_stokes/fda_benchmark/include/grid.h:390]: Why isn't there a Triangulation in the red line?

**Comment by nfehn** [comment]
I still have to go through more directories in `include/exadg` ...

File: applications/fluid_structure_interaction/pressure_wave/application.h [hunk 2]
```diff
@@ -356,9 +356,9 @@ class Application : public FluidFSI::ApplicationBase<dim, Number>
           {
             if(cell->manifold_id() == manifold_ids[i])
             {
-              manifold_vec[i] =
-                std::shared_ptr<dealii::Manifold<dim>>(static_cast<dealii::Manifold<dim> *>(
-                  new OneSidedCylindricalManifold<dim>(cell, face_ids[i], dealii::Point<dim>())));
+              manifold_vec[i] = std::shared_ptr<dealii::Manifold<dim>>(
+                static_cast<dealii::Manifold<dim> *>(new OneSidedCylindricalManifold<dim>(
+                  tria, cell, face_ids[i], dealii::Point<dim>())));
```
[the remark this answers is not in the document]
Comment (nfehn) [reply applications/fluid_structure_interaction/pressure_wave/application.h:361]: I meantioned this in issue #143.

**Pull request merged**